compose/snippets/src/main/java/com/example/compose/snippets/components/Menus.kt

```diff
@@ -0,0 +1,217 @@
+                Text("Dropdown menu with many items")
+            }
+            Button(onClick = { currentExample = { DropdownMenuWithDetails() } }) {
+                Text("Dropdown menu with sections and icons")
+            }
+        }
+    }
+}
+
+// [START android_compose_components_minimaldropdownmenu]
+@Composable
+fun MinimalDropdownMenu() {
+    var expanded by remember { mutableStateOf(false) }
+    Box(
+        modifier = Modifier
+            .padding(16.dp)
+    ) {
+        IconButton(onClick = { expanded = !expanded }) {
+            Icon(Icons.Default.MoreVert, contentDescription = "More options")
+        }
+        DropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false }
+        ) {
+            DropdownMenuItem(
+                text = { Text("Option 1") },
+                onClick = { /* Do something... */ }
+            )
+            DropdownMenuItem(
+                text = { Text("Option 2") },
+                onClick = { /* Do something... */ }
+            )
+        }
+    }
+}
+// [END android_compose_components_minimaldropdownmenu]
+
+@Preview
+@Composable
+fun MinimalDropdownMenuPreview() {
+    MinimalDropdownMenu()
+}
+
+// [START android_compose_components_longbasicdropdownmenu]
+@Composable
+fun LongBasicDropdownMenu() {
+    var expanded by remember { mutableStateOf(false) }
+    // Placeholder list of 100 strings for demonstration
+    val menuItemData = List(100) { "Option ${it + 1}" }
+
+    Box(
+        modifier = Modifier
+            .padding(16.dp)
+    ) {
+        IconButton(onClick = { expanded = !expanded }) {
+            Icon(Icons.Default.MoreVert, contentDescription = "More options")
+        }
+        DropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false }
+        ) {
+            menuItemData.forEach { option ->
+                DropdownMenuItem(
+                    text = { Text(option) },
+                    onClick = { /* Do something... */ }
+                )
+            }
+        }
+    }
+}
+// [END android_compose_components_longbasicdropdownmenu]
+
+@Preview
+@Composable
+fun LongBasicDropdownMenuPreview() {
+    LongBasicDropdownMenu()
+}
+
+// [START android_compose_components_dropdownmenuwithdetails]
+@Composable
+fun DropdownMenuWithDetails() {
+    var expanded by remember { mutableStateOf(false) }
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(16.dp)
+    ) {
+        IconButton(onClick = { expanded = !expanded }) {
+            Icon(Icons.Default.MoreVert, contentDescription = "More options")
+        }
+        DropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false }
+        ) {
+            // First section
+            DropdownMenuItem(
+                text = { Text("Profile") },
+                leadingIcon = { Icon(Icons.Default.Person, contentDescription = null) },
```

Comment:
Please use outlined icons to make the dropdown feel less heavy

Reply:
Done.